lib/icinga/commandutils.cpp

```diff
@@ -0,0 +1,73 @@
+#include "icinga/commandutils.hpp"
+#include "icinga/host.hpp"
+#include "icinga/checkcommand.hpp"
+#include "remote/apilistener.hpp"
+#include "remote/httputility.hpp"
+
+using namespace icinga;
+
+
+void CommandUtils::ExecuteCommandLocally(const Checkable::Ptr& checkable, const CheckCommand::Ptr& command, const Dictionary::Ptr& params)
+{
+
+    Dictionary::Ptr macros;
+    if (!params->Contains("macros"))
+        macros = new Dictionary();
+    else
+        macros = params->Get("macros");
+
+    Array::Ptr command_line_tmp = command->GetCommandLine();
+    Value shutdown_command_value = params->Get("shutdown_command");
```

Comment:
This value is retrieved in various places, but it never gets set. Is that a new command object attribute or is the PR incomplete?

Reply:
The endpoint is called by passing a parameter called shutdown_command, which represents a valid shutdown command for the target host. An example follows:

`curl -s -k -u root:admin -H 'X-HTTP-Method-Override: POST' -H 'Accept: application/json' https://localhost:5665/v1/actions/shutdown-host -d '{ "type": "Host", "filter": "host.name==\"<hostname>\"", "shutdown_command": ["systemctl", "poweroff"]}'
`